src/Microsoft.OData.Edm/Schema/EdmSingleton.cs

```diff
@@ -29,7 +29,7 @@ public EdmSingleton(IEdmEntityContainer container, string name, IEdmEntityType e
 
             this.container = container;
             this.entityType = entityType;
-            this.path = new EdmPathExpression(name);
+            this.path = new EdmPathExpression(this.container.FullName() + "/" + name);

```

Comment:
>this.container.FullName() + "/" + name [](start = 46, length = 38)

use string.Concat()

Reply:
Not applicable

---
In reply to: [542982068](https://github.com/OData/odata.net/pull/1950#discussion_r542982068) [](ancestors = 542982068)